### PR TITLE
Remove routes that override GC Articles routes

### PR DESCRIPTION
**Pull request by EliseKa**
# Summary | Résumé

Removed the routes for the following urls:
/privacy
/guidance
/security
/features
/why-notify

[Link to ZenHub card](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/500)

# Test instructions | Instructions pour tester la modification

1. Paste each of the following urls in your browser:
- https://articles.cdssandbox.xyz/notification-gc-notify/privacy/
- https://articles.cdssandbox.xyz/notification-gc-notify/guidance/
- https://articles.cdssandbox.xyz/notification-gc-notify/security/
- https://articles.cdssandbox.xyz/notification-gc-notify/features/
- https://articles.cdssandbox.xyz/notification-gc-notify/why-notify/
2. Toggle to French
3. Assess if the url is now the French url as it should be, or if it stayed the English one (which means the old route still takes precedence)

### CODE REVIEW

**Comment by jimleroyer** [comment]
Errors from the heroku instance. It seems the deleted pages are referenced and it can't find the deleted paths now:

```
2022-04-07T22:33:10.979604+00:00 app[web.1]: {"name": "app", "levelname": "ERROR", "message": "500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.", "pathname": "/app/app/__init__.py", "lineno": 743, "exc_info": "Traceback (most recent call last):\n  File \"/app/app/articles/__init__.py\", line 108, in request_content\n    raise NotFound()\nwerkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/app.py\", line 1950, in full_dispatch_request\n    rv = self.dispatch_request()\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/app.py\", line 1936, in dispatch_request\n    return self.view_functions[rule.endpoint](**req.view_args)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py\", line 50, in _nr_wrapper_handler_\n    return wrapped(*args, **kwargs)\n  File \"/app/app/main/views/index.py\", line 361, in page_content\n    response = request_content(f\"wp/v2/pages/{page_id}\", {\"slug\": path}, auth_required=auth_required)\n  File \"/app/app/articles/__init__.py\", line 118, in request_content\n    abort(404)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/werkzeug/exceptions.py\", line 822, in abort\n    return _aborter(status, *args, **kwargs)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/werkzeug/exceptions.py\", line 807, in __call__\n    raise self.mapping[code](*args, **kwargs)\nwerkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/app.py\", line 2447, in wsgi_app\n    response = self.full_dispatch_request()\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/app.py\", line 1952, in full_dispatch_request\n    rv = self.handle_user_exception(e)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py\", line 111, in _nr_wrapper_Flask_handle_exception_\n    return wrapped(*args, **kwargs)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/app.py\", line 1816, in handle_user_exception\n    return self.handle_http_exception(e)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py\", line 94, in _nr_wrapper_Flask_handle_http_exception_\n    return wrapped(*args, **kwargs)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/app.py\", line 1744, in handle_http_exception\n    return handler(e)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py\", line 129, in _nr_wrapper_error_handler_\n    return wrapped(*args, **kwargs)\n  File \"/app/app/__init__.py\", line 682, in handle_not_found\n    return _error_response(404)\n  File \"/app/app/__init__.py\", line 633, in _error_response\n    resp = make_response(render_template(\"error/{0}.html\".format(error_code)), error_code)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py\", line 150, in literal_wrapper\n    return wrapped(*args, **kwargs)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/templating.py\", line 137, in render_template\n    return _render(\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/templating.py\", line 120, in _render\n    rv = template.render(context)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py\", line 93, in dynamic_wrapper\n    return wrapped(*args, **kwargs)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/jinja2/environment.py\", line 1090, in render\n    self.environment.handle_exception()\n  File \"/app/.heroku/python/lib/python3.9/site-packages/jinja2/environment.py\", line 832, in handle_exception\n    reraise(*rewrite_traceback_stack(source=source))\n  File \"/app/.heroku/python/lib/python3.9/site-packages/jinja2/_compat.py\", line 28, in reraise\n    raise value.with_traceback(tb)\n  File \"/app/app/templates/error/404.html\", line 1, in top-level template code\n    {% extends  \"admin_template.html\" %}\n  File \"/app/app/templates/admin_template.html\", line 9, in top-level template code\n    {% set extra_title = ' - ' + current_service.name if current_service.name and current_user.has_permissions() else ''  %}\n  File \"/app/app/templates/main_template.html\", line 115, in top-level template code\n    {% block gc_header %}\n  File \"/app/app/templates/main_template.html\", line 129, in block \"gc_header\"\n    {% include 'partials/nav/gc_header_nav.html' %}\n  File \"/app/app/templates/partials/nav/gc_header_nav.html\", line 34, in top-level template code\n    {% include 'partials/nav/gc_header_nav_mobile.html' %}\n  File \"/app/app/templates/partials/nav/gc_header_nav_mobile.html\", line 41, in top-level template code\n    {{ nav_menu_item_mobile(url_for('main.why-notify'),_('Why GC Notify'),header_navigation.is_selected('why-notify')) }}\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/helpers.py\", line 370, in url_for\n    return appctx.app.handle_url_build_error(error, endpoint, values)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/app.py\", line 2216, in handle_url_build_error\n    reraise(exc_type, exc_value, tb)\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/_compat.py\", line 39, in reraise\n    raise value\n  File \"/app/.heroku/python/lib/python3.9/site-packages/flask/helpers.py\", line 357, in url_for\n    rv = url_adapter.build(\n  File \"/app/.heroku/python/lib/python3.9/site-packages/werkzeug/routing.py\", line 2179, in build\n    raise BuildError(endpoint, values, method, self)\nwerkzeug.routing.BuildError: Could not build url for endpoint 'main.why-notify'. Did you mean 'main.verify' instead?", "time": "2022-04-07T22:33:10", "requestId": null, "application": "admin", "logType": "application"}
2022-04-07T22:33:11.090962+00:00 app[web.1]: [2022-04-07 22:33:11 +0000] [23] [ERROR] Error handling request /favicon.ico
2022-04-07T22:33:11.090966+00:00 app[web.1]: Traceback (most recent call last):
2022-04-07T22:33:11.090966+00:00 app[web.1]:   File "/app/app/articles/__init__.py", line 108, in request_content
2022-04-07T22:33:11.090967+00:00 app[web.1]:     raise NotFound()
2022-04-07T22:33:11.090968+00:00 app[web.1]: werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
2022-04-07T22:33:11.090968+00:00 app[web.1]: 
2022-04-07T22:33:11.090969+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2022-04-07T22:33:11.090969+00:00 app[web.1]: 
2022-04-07T22:33:11.090969+00:00 app[web.1]: Traceback (most recent call last):
2022-04-07T22:33:11.090970+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
2022-04-07T22:33:11.090970+00:00 app[web.1]:     rv = self.dispatch_request()
2022-04-07T22:33:11.090971+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
2022-04-07T22:33:11.090971+00:00 app[web.1]:     return self.view_functions[rule.endpoint](**req.view_args)
2022-04-07T22:33:11.090972+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py", line 50, in _nr_wrapper_handler_
2022-04-07T22:33:11.090972+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.090973+00:00 app[web.1]:   File "/app/app/main/views/index.py", line 361, in page_content
2022-04-07T22:33:11.090973+00:00 app[web.1]:     response = request_content(f"wp/v2/pages/{page_id}", {"slug": path}, auth_required=auth_required)
2022-04-07T22:33:11.090973+00:00 app[web.1]:   File "/app/app/articles/__init__.py", line 118, in request_content
2022-04-07T22:33:11.090974+00:00 app[web.1]:     abort(404)
2022-04-07T22:33:11.090974+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/werkzeug/exceptions.py", line 822, in abort
2022-04-07T22:33:11.090974+00:00 app[web.1]:     return _aborter(status, *args, **kwargs)
2022-04-07T22:33:11.090975+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/werkzeug/exceptions.py", line 807, in __call__
2022-04-07T22:33:11.090975+00:00 app[web.1]:     raise self.mapping[code](*args, **kwargs)
2022-04-07T22:33:11.090975+00:00 app[web.1]: werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
2022-04-07T22:33:11.090976+00:00 app[web.1]: 
2022-04-07T22:33:11.090976+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2022-04-07T22:33:11.090976+00:00 app[web.1]: 
2022-04-07T22:33:11.090976+00:00 app[web.1]: Traceback (most recent call last):
2022-04-07T22:33:11.090977+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
2022-04-07T22:33:11.090978+00:00 app[web.1]:     response = self.full_dispatch_request()
2022-04-07T22:33:11.090978+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
2022-04-07T22:33:11.090978+00:00 app[web.1]:     rv = self.handle_user_exception(e)
2022-04-07T22:33:11.090979+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py", line 111, in _nr_wrapper_Flask_handle_exception_
2022-04-07T22:33:11.090979+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.090979+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 1816, in handle_user_exception
2022-04-07T22:33:11.090980+00:00 app[web.1]:     return self.handle_http_exception(e)
2022-04-07T22:33:11.090995+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py", line 94, in _nr_wrapper_Flask_handle_http_exception_
2022-04-07T22:33:11.090996+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.090996+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 1744, in handle_http_exception
2022-04-07T22:33:11.090996+00:00 app[web.1]:     return handler(e)
2022-04-07T22:33:11.090997+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py", line 129, in _nr_wrapper_error_handler_
2022-04-07T22:33:11.090998+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.090998+00:00 app[web.1]:   File "/app/app/__init__.py", line 682, in handle_not_found
2022-04-07T22:33:11.090998+00:00 app[web.1]:     return _error_response(404)
2022-04-07T22:33:11.090999+00:00 app[web.1]:   File "/app/app/__init__.py", line 633, in _error_response
2022-04-07T22:33:11.090999+00:00 app[web.1]:     resp = make_response(render_template("error/{0}.html".format(error_code)), error_code)
2022-04-07T22:33:11.091000+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 150, in literal_wrapper
2022-04-07T22:33:11.091000+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.091000+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/templating.py", line 137, in render_template
2022-04-07T22:33:11.091001+00:00 app[web.1]:     return _render(
2022-04-07T22:33:11.091001+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/templating.py", line 120, in _render
2022-04-07T22:33:11.091001+00:00 app[web.1]:     rv = template.render(context)
2022-04-07T22:33:11.091001+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 93, in dynamic_wrapper
2022-04-07T22:33:11.091002+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.091002+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/jinja2/environment.py", line 1090, in render
2022-04-07T22:33:11.091002+00:00 app[web.1]:     self.environment.handle_exception()
2022-04-07T22:33:11.091003+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/jinja2/environment.py", line 832, in handle_exception
2022-04-07T22:33:11.091003+00:00 app[web.1]:     reraise(*rewrite_traceback_stack(source=source))
2022-04-07T22:33:11.091003+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/jinja2/_compat.py", line 28, in reraise
2022-04-07T22:33:11.091004+00:00 app[web.1]:     raise value.with_traceback(tb)
2022-04-07T22:33:11.091009+00:00 app[web.1]:   File "/app/app/templates/error/404.html", line 1, in top-level template code
2022-04-07T22:33:11.091010+00:00 app[web.1]:     {% extends  "admin_template.html" %}
2022-04-07T22:33:11.091010+00:00 app[web.1]:   File "/app/app/templates/admin_template.html", line 9, in top-level template code
2022-04-07T22:33:11.091010+00:00 app[web.1]:     {% set extra_title = ' - ' + current_service.name if current_service.name and current_user.has_permissions() else ''  %}
2022-04-07T22:33:11.091011+00:00 app[web.1]:   File "/app/app/templates/main_template.html", line 115, in top-level template code
2022-04-07T22:33:11.091011+00:00 app[web.1]:     {% block gc_header %}
2022-04-07T22:33:11.091011+00:00 app[web.1]:   File "/app/app/templates/main_template.html", line 129, in block "gc_header"
2022-04-07T22:33:11.091012+00:00 app[web.1]:     {% include 'partials/nav/gc_header_nav.html' %}
2022-04-07T22:33:11.091012+00:00 app[web.1]:   File "/app/app/templates/partials/nav/gc_header_nav.html", line 34, in top-level template code
2022-04-07T22:33:11.091012+00:00 app[web.1]:     {% include 'partials/nav/gc_header_nav_mobile.html' %}
2022-04-07T22:33:11.091012+00:00 app[web.1]:   File "/app/app/templates/partials/nav/gc_header_nav_mobile.html", line 41, in top-level template code
2022-04-07T22:33:11.091013+00:00 app[web.1]:     {{ nav_menu_item_mobile(url_for('main.why-notify'),_('Why GC Notify'),header_navigation.is_selected('why-notify')) }}
2022-04-07T22:33:11.091013+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/helpers.py", line 370, in url_for
2022-04-07T22:33:11.091014+00:00 app[web.1]:     return appctx.app.handle_url_build_error(error, endpoint, values)
2022-04-07T22:33:11.091014+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 2216, in handle_url_build_error
2022-04-07T22:33:11.091014+00:00 app[web.1]:     reraise(exc_type, exc_value, tb)
2022-04-07T22:33:11.091014+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
2022-04-07T22:33:11.091015+00:00 app[web.1]:     raise value
2022-04-07T22:33:11.091015+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/helpers.py", line 357, in url_for
2022-04-07T22:33:11.091015+00:00 app[web.1]:     rv = url_adapter.build(
2022-04-07T22:33:11.091015+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/werkzeug/routing.py", line 2179, in build
2022-04-07T22:33:11.091016+00:00 app[web.1]:     raise BuildError(endpoint, values, method, self)
2022-04-07T22:33:11.091016+00:00 app[web.1]: werkzeug.routing.BuildError: Could not build url for endpoint 'main.why-notify'. Did you mean 'main.verify' instead?
2022-04-07T22:33:11.091016+00:00 app[web.1]: 
2022-04-07T22:33:11.091017+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2022-04-07T22:33:11.091017+00:00 app[web.1]: 
2022-04-07T22:33:11.091017+00:00 app[web.1]: Traceback (most recent call last):
2022-04-07T22:33:11.091017+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/gunicorn/workers/base_async.py", line 55, in handle
2022-04-07T22:33:11.091018+00:00 app[web.1]:     self.handle_request(listener_name, req, client, addr)
2022-04-07T22:33:11.091018+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/gunicorn/workers/base_async.py", line 108, in handle_request
2022-04-07T22:33:11.091019+00:00 app[web.1]:     respiter = self.wsgi(environ, resp.start_response)
2022-04-07T22:33:11.091019+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/wsgi_application.py", line 682, in _nr_wsgi_application_wrapper_
2022-04-07T22:33:11.091019+00:00 app[web.1]:     result = wrapped(environ, _start_response)
2022-04-07T22:33:11.091020+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/integrations/flask.py", line 86, in sentry_patched_wsgi_app
2022-04-07T22:33:11.091020+00:00 app[web.1]:     return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
2022-04-07T22:33:11.091021+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/integrations/wsgi.py", line 140, in __call__
2022-04-07T22:33:11.091021+00:00 app[web.1]:     reraise(*_capture_exception(hub))
2022-04-07T22:33:11.091021+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/_compat.py", line 54, in reraise
2022-04-07T22:33:11.091022+00:00 app[web.1]:     raise value
2022-04-07T22:33:11.091022+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/integrations/wsgi.py", line 133, in __call__
2022-04-07T22:33:11.091022+00:00 app[web.1]:     rv = self.app(
2022-04-07T22:33:11.091022+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/integrations/flask.py", line 86, in <lambda>
2022-04-07T22:33:11.091022+00:00 app[web.1]:     return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
2022-04-07T22:33:11.091023+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 2464, in __call__
2022-04-07T22:33:11.091023+00:00 app[web.1]:     return self.wsgi_app(environ, start_response)
2022-04-07T22:33:11.091023+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/notifications_utils/request_helper.py", line 74, in __call__
2022-04-07T22:33:11.091024+00:00 app[web.1]:     return self.app(environ, rewrite_response_headers)
2022-04-07T22:33:11.091024+00:00 app[web.1]:   File "/app/app/proxy_fix.py", line 11, in __call__
2022-04-07T22:33:11.091024+00:00 app[web.1]:     return self.app(environ, start_response)
2022-04-07T22:33:11.091024+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/werkzeug/middleware/proxy_fix.py", line 169, in __call__
2022-04-07T22:33:11.091025+00:00 app[web.1]:     return self.app(environ, start_response)
2022-04-07T22:33:11.091025+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/werkzeug/middleware/proxy_fix.py", line 169, in __call__
2022-04-07T22:33:11.091025+00:00 app[web.1]:     return self.app(environ, start_response)
2022-04-07T22:33:11.091026+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/wsgi_application.py", line 682, in _nr_wsgi_application_wrapper_
2022-04-07T22:33:11.091026+00:00 app[web.1]:     result = wrapped(environ, _start_response)
2022-04-07T22:33:11.091029+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 2450, in wsgi_app
2022-04-07T22:33:11.091030+00:00 app[web.1]:     response = self.handle_exception(e)
2022-04-07T22:33:11.091030+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py", line 111, in _nr_wrapper_Flask_handle_exception_
2022-04-07T22:33:11.091030+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.091030+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 1879, in handle_exception
2022-04-07T22:33:11.091031+00:00 app[web.1]:     server_error = handler(server_error)
2022-04-07T22:33:11.091031+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py", line 129, in _nr_wrapper_error_handler_
2022-04-07T22:33:11.091031+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.091031+00:00 app[web.1]:   File "/app/app/__init__.py", line 750, in handle_bad_request
2022-04-07T22:33:11.091032+00:00 app[web.1]:     return _error_response(500)
2022-04-07T22:33:11.091032+00:00 app[web.1]:   File "/app/app/__init__.py", line 633, in _error_response
2022-04-07T22:33:11.091033+00:00 app[web.1]:     resp = make_response(render_template("error/{0}.html".format(error_code)), error_code)
2022-04-07T22:33:11.091033+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 150, in literal_wrapper
2022-04-07T22:33:11.091033+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.091033+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/templating.py", line 137, in render_template
2022-04-07T22:33:11.091034+00:00 app[web.1]:     return _render(
2022-04-07T22:33:11.091034+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/templating.py", line 120, in _render
2022-04-07T22:33:11.091034+00:00 app[web.1]:     rv = template.render(context)
2022-04-07T22:33:11.091034+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 93, in dynamic_wrapper
2022-04-07T22:33:11.091035+00:00 app[web.1]:     return wrapped(*args, **kwargs)
2022-04-07T22:33:11.091035+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/jinja2/environment.py", line 1090, in render
2022-04-07T22:33:11.091035+00:00 app[web.1]:     self.environment.handle_exception()
2022-04-07T22:33:11.091035+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/jinja2/environment.py", line 832, in handle_exception
2022-04-07T22:33:11.091036+00:00 app[web.1]:     reraise(*rewrite_traceback_stack(source=source))
2022-04-07T22:33:11.091036+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/jinja2/_compat.py", line 28, in reraise
2022-04-07T22:33:11.091036+00:00 app[web.1]:     raise value.with_traceback(tb)
2022-04-07T22:33:11.091036+00:00 app[web.1]:   File "/app/app/templates/error/500.html", line 1, in top-level template code
2022-04-07T22:33:11.091037+00:00 app[web.1]:     {% extends  "admin_template.html" %}
2022-04-07T22:33:11.091037+00:00 app[web.1]:   File "/app/app/templates/admin_template.html", line 9, in top-level template code
2022-04-07T22:33:11.091038+00:00 app[web.1]:     {% set extra_title = ' - ' + current_service.name if current_service.name and current_user.has_permissions() else ''  %}
2022-04-07T22:33:11.091038+00:00 app[web.1]:   File "/app/app/templates/main_template.html", line 115, in top-level template code
2022-04-07T22:33:11.091038+00:00 app[web.1]:     {% block gc_header %}
2022-04-07T22:33:11.091039+00:00 app[web.1]:   File "/app/app/templates/main_template.html", line 129, in block "gc_header"
2022-04-07T22:33:11.091039+00:00 app[web.1]:     {% include 'partials/nav/gc_header_nav.html' %}
2022-04-07T22:33:11.091039+00:00 app[web.1]:   File "/app/app/templates/partials/nav/gc_header_nav.html", line 34, in top-level template code
2022-04-07T22:33:11.091039+00:00 app[web.1]:     {% include 'partials/nav/gc_header_nav_mobile.html' %}
2022-04-07T22:33:11.091040+00:00 app[web.1]:   File "/app/app/templates/partials/nav/gc_header_nav_mobile.html", line 41, in top-level template code
2022-04-07T22:33:11.091040+00:00 app[web.1]:     {{ nav_menu_item_mobile(url_for('main.why-notify'),_('Why GC Notify'),header_navigation.is_selected('why-notify')) }}
2022-04-07T22:33:11.091040+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/helpers.py", line 370, in url_for
2022-04-07T22:33:11.091041+00:00 app[web.1]:     return appctx.app.handle_url_build_error(error, endpoint, values)
2022-04-07T22:33:11.091041+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/app.py", line 2216, in handle_url_build_error
2022-04-07T22:33:11.091041+00:00 app[web.1]:     reraise(exc_type, exc_value, tb)
2022-04-07T22:33:11.091042+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
2022-04-07T22:33:11.091042+00:00 app[web.1]:     raise value
2022-04-07T22:33:11.091042+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/flask/helpers.py", line 357, in url_for
2022-04-07T22:33:11.091042+00:00 app[web.1]:     rv = url_adapter.build(
2022-04-07T22:33:11.091043+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.9/site-packages/werkzeug/routing.py", line 2179, in build
2022-04-07T22:33:11.091043+00:00 app[web.1]:     raise BuildError(endpoint, values, method, self)
2022-04-07T22:33:11.091044+00:00 app[web.1]: werkzeug.routing.BuildError: Could not build url for endpoint 'main.why-notify'. Did you mean 'main.verify' instead?
```

File: app/main/views/index.py [hunk 1]
```diff
@@ -91,7 +91,35 @@ def verify_mobile():
...
+            "views/page-content.html",
+            title=title,
+            html_content=html_content,
+            nav_items=nav_items,
+            slug=slug_en,
+            lang_url=get_lang_url(response, bool(page_id)),
+            stats=get_latest_stats(get_current_locale(current_app)) if slug_en == "home" else None,
+        )
+    else:
+        abort(404)
```
Comment (amazingphilippe) [inline app/main/views/index.py:122]: This is for when the GC Articles url is the same as a route that already existed. This is the one that I'd like to reduce repetition, I'm using that same wall of code for a few routes. 

1. I request content for that slug
2. I think if a string comes back, it means we are requesting it for the wrong lang: redirect to the FR url
3. Else, render just like the wildcard route.

File: app/main/views/index.py [hunk 2]
```diff
@@ -291,27 +374,92 @@ def features_letters():
...
 
 
 @main.route("/format", endpoint="format")
 def format():
-    return render_template("views/guidance/format.html")
+    if get_current_locale(current_app) == "en":
+        return redirect("formatting-guide", 301)
+    else:
+        return redirect("guide-mise-en-forme", 301)
```
Comment (amazingphilippe) [inline app/main/views/index.py:413]: This is for the rest, when routes aren't conflicting. This allows us not to worry about the links within the app. Like in the footer, within forms...

File: app/main/views/index.py (hunk 1 above)
Comment (patheard) [reply app/main/views/index.py:122]: That makes sense - we can definitely cut down on the repetition be creating two new functions:
1. one for routes that are conflicting, like `/privacy`, and
2. another for the non-clicking routes like `/format`.

I can walk through it with you when we chat later this morning!

File: app/main/views/index.py [hunk 3]
```diff
@@ -274,19 +372,94 @@ def features_letters():
...
+            "views/page-content.html",
+            title=title,
+            html_content=html_content,
+            nav_items=nav_items,
+            slug=slug_en,
+            lang_url=get_lang_url(response, bool(page_id)),
+            stats=get_latest_stats(get_current_locale(current_app)) if slug_en == "home" else None,
+        )
+    else:
+        abort(404)
```
Comment (jimleroyer) [inline app/main/views/index.py:454]: A lot of the code is repeating and it feels we could reuse that logic for each redirection, right?

Comment (jimleroyer) [reply app/main/views/index.py:454]: Sorry I just left that comment but I wrote it yesterday evening. I see that some discussion is taking place to this effect. 👍

File: app/templates/views/privacy.html [hunk 4]
```diff
@@ -8,6 +8,7 @@
   <div class="grid-row contain-floats">
     <div class="md:w-2/3 float-left py-0 px-0 px-gutterHalf box-border">
       <h1 class="heading-large">{{ _('Privacy') }}</h1>
+      Helloooo!
```
Comment (patheard) [inline app/templates/views/privacy.html:11]: Our test change snuck in :)

File: app/articles/__init__.py [hunk 5]
```diff
@@ -155,6 +156,38 @@ def _get_nav_wp(locale: str) -> Optional[list]:
     return nav_items
 
 
+def get_content_by_slug_or_redirect(slug_en, slug_fr):
```
Comment (patheard) [inline app/articles/__init__.py:159]: It would be nice if we could add a few unit tests for this new function.  Happy to pair and help write the tests.

Comment (amazingphilippe) [reply app/articles/__init__.py:159]: +1, I also noticed we don't test the wildcard route. I think these could have similar tests.

File: app/templates/views/privacy.html (hunk 4 above)
Comment (amazingphilippe) [reply app/templates/views/privacy.html:11]: ooops!

File: tests/app/main/views/test_index.py [hunk 6]
```diff
@@ -167,6 +163,21 @@ def test_old_static_pages_redirect(client, view, expected_view):
...
+        ("messages_status", "message-delivery-status"),
+        ("format", "formatting-guide"),
+        ("personalise", "personalisation-guide"),
+    ]
+)
+def test_old_static_pages_redirect_to_wildcard(client, view, expected_redirect_url):
+    response = client.get(url_for("main.{}".format(view)))
+    assert response.status_code == 301
+    assert response.location.endswith(f"/{expected_redirect_url}")
+
```
Comment (amazingphilippe) [inline tests/app/main/views/test_index.py:179]: So here, I decided I wanted to tests the redirects specifically. So I moved some parameters further above to this new test. I'm thinking we should also test the en-fr redirects in another test. That'd be nice.

File: app/articles/__init__.py (hunk 5 above)
Comment (timarney) [reply app/articles/__init__.py:159]: @dsamojlenko has modified a lot of this logic in PR 

To work in a bunch of caching + simplify things overall.  That in turn will end up modifying the test so would suggest moving forward as is on the testing front and look to get Dave's PR tested and merged.

https://github.com/cds-snc/notification-admin/pull/1240

**Comment by jimleroyer** [comment]
First time I clicked on a review environment link with AWS env @patheard , it's great! 😄

**Comment by jimleroyer** [comment]
Closing this PR as the effort has been continued into https://github.com/cds-snc/notification-admin/pull/1275

**Pull request closed without merging**